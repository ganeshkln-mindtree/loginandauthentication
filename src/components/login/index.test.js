import React from 'react';
import { shallow } from 'enzyme';
import Login from '../login';
import { shallowToJson } from 'enzyme-to-json';

const setup = propsoverrides => {
  const component = shallow(<Login />);
  return { component };
};

describe('test index', () => {
  const { component } = setup();
  it('test case', () => {
    expect(shallowToJson(component)).toMatchSnapshot();
  });
  it('test case 3', () => {
    expect(component.exists()).toBe(true);
  });
  it('test case 2', () => {
    const spy = jest.spyOn(component.instance(), 'handleSubmit');
    component.instance().forceUpdate();
    component
      .find('button')
      .simulate('submit', { preventDefault() {} })
      .first();
    component.instance().handleSubmit({ preventDefault() {} });
    //console.log(handleSubmit.mock.calls.length);

    expect(spy).toHaveBeenCalled();
  });
});
