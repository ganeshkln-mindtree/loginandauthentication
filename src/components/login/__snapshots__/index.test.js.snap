// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test index test case 1`] = `
<form
  noValidate={true}
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <label
      for="emailaddress"
    >
      Email address
    </label>
    <input
      aria-describedby="emailHelp"
      className="form-control"
      id="emailaddress"
      placeholder="Enter email"
      type="email"
    />
    <small
      className="form-text text-muted"
      id="emailHelp"
    >
      We'll never share your email with anyone else.
    </small>
  </div>
  <div
    className="form-group"
  >
    <label
      for="password"
    >
      Password
    </label>
    <input
      className="form-control"
      id="password"
      placeholder="Password"
      type="password"
    />
  </div>
  <div
    className="form-group form-check"
  >
    <Link
      to="/sign-up"
    >
      New User
       
    </Link>
  </div>
  <button
    className="btn btn-primary"
    type="submit"
  >
    Submit
  </button>
</form>
`;
